src/sql/platform/connection/common/utils.ts

```diff
@@ -137,3 +139,42 @@ export function isServerConnection(profile: IConnectionProfile): boolean {
 	// If the user did not specify a database in the original connection, then this is considered a server-level connection
 	return !profile.options.originalDatabase;
 }
+
+/**
+ * Convert a IConnectionProfile with services to an azdata.connection.ConnectionProfile
+ * shaped object that can be sent via RPC.
+ * @param profile The profile to be converted.
+ * @param deepCopyOptions whether to deep copy the options or not.
+ * @param removeFunction the function that strips the credentials from the connection profile if provided.
+ * @returns An azdata.connection.ConnectionProfile shaped object that contains only the data and none of the services.
+ */
+export function convertToRpcConnectionProfile(profile: IConnectionProfile | undefined, deepCopyOptions: boolean = false, removeFunction?: (profile: IConnectionProfile) => IConnectionProfile): sqlExtHostTypes.ConnectionProfile {
```

Comment:
I missed updating the return type in my last comment too so that it matches the actual types possibly returned.

```suggestion
export function convertToRpcConnectionProfile(profile: IConnectionProfile | undefined, deepCopyOptions: boolean = false, removeFunction?: (profile: IConnectionProfile) => IConnectionProfile): sqlExtHostTypes.ConnectionProfile | undefined {
```